crates/btc-relay/src/lib.rs

```diff
@@ -483,6 +478,28 @@ impl<T: Config> Module<T> {
         Ok(())
     }
 
+    pub fn _verify_and_validate_transaction(
+        raw_merkle_proof: Vec<u8>,
+        raw_tx: Vec<u8>,
+        recipient_btc_address: BtcAddress,
+        minimum_btc: Option<i64>,
+        op_return_id: Option<H256>,
+        confirmations: Option<u32>,
+    ) -> Result<(BtcAddress, i64), DispatchError> {
+        let transaction = Self::parse_transaction(&raw_tx)?;
+
+        // Verify that the transaction is indeed included in the main chain
+        Self::_verify_transaction_inclusion(transaction.tx_id(), raw_merkle_proof, confirmations)?;
```

Comment:
No change is needed, the `tx_id()` function is already correct. The problem was previously we weren't using it - we were using `sha256d_le` directly on the raw tx. We correctness of the txid in the presence of witness data was already covered by the tests, but I added two tests where the difference is extra explicit.

As for the `wtxid()` function, it would be easily implementable, but bitcoin-cli does not expose `wtxid` except for in mempool data, so it'd be a bit hard to check if any test of `wtxid()` would be correct. Since we don't use `wtxid` anywhere anyway, I don't think it's not worth the effort to add it now